src/System.AppContext/src/System/AppContext.cs

```diff
@@ -18,8 +18,7 @@ private enum SwitchValueState
             UnknownValue = 0x8 // Not used on .NET Native
         }
 
-        private static Dictionary<string, SwitchValueState> s_switchMap = new Dictionary<string, SwitchValueState>();
-        private static readonly object s_syncLock = new object();
+        private static readonly Dictionary<string, SwitchValueState> s_switchMap = new Dictionary<string, SwitchValueState>();
```

Comment:
We need to know if the value that we retrieve was retrieved via an override or if it came from the default value.

[This](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/AppContext/AppContext.cs#L57) is the implementation on Full Framework if you want to take a look.

I am going to file an issue tracking this inside the CoreCLR repo.